Do not attempt to manage home directories.

rhass opened the pull request:
This resolves a bug in the `gemrc` provider when using `name :local` or `gemrc "local"` in which the resource could potentially recursively change file permissions for the home directory. We now skip the
`directory` resource call within the `gemrc` custom resource when the gemrc resource name resolves to either `:local` or `"local"` to ensure files and directories do not become tainted when the user or group
values are set.

Additionally, this cleans up the chefspec slightly and adds a context to make it a little cleaner.

/cc @chef-cookbooks/engineering-services @tas50

Commit:
Do not attempt to manage home directories.

This resolves a bug in the `gemrc` provider when using `name :local` or
`gemrc "local"` in which the resource could potentially recursively
change file permissions for the home directory. We now skip the
`directory` resource call within the `gemrc` custom resource when the
gemrc resource name resolves to either `:local` or `"local"` to ensure
files and directories do not become tainted when the user or group
values are set.

Additionally, this cleans up the chefspec slightly and adds a context
to make it a little cleaner.

Signed-off-by: Ryan Hass <rhass@chef.io>

schisamo commented on resources/gemrc.rb:
```diff
@@ -46,6 +46,7 @@
     user new_resource.user if new_resource.property_is_set?("user")
     group new_resource.group if new_resource.property_is_set?("group")
     recursive true
+    not_if { new_resource.name == :local || new_resource.name == "local" }
```
This can be simplified:

``` ruby
not_if { new_resource.name.to_sym == :local }
```

Ditto on the `coerce_path` method:

``` ruby
def coerce_path(value)
  case value.to_sym
  when :local
    ::File.join(Dir.home, ".gemrc")
  when :global
    Gem::ConfigFile::SYSTEM_WIDE_CONFIG_FILE
  else
    value
  end
end
```

Commit:
Simplify conditional checks.

This converts string values to symbols to simplify the conditials for
path coercion and directory creation.

scotthain reviewed (approved):
Looks good!
